llvm/lib/SYCLLowerIR/LowerWGLocalMemory.cpp

```diff
@@ -0,0 +1,125 @@
+//===-- LowerWGLocalMemory.cpp - SYCL kernel local memory allocation pass -===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass replaces calls to __sycl_allocateLocalMemory(Size, Alignment)
+// function with allocation of memory in local address space at the kernel
+// scope.
```

Comment:
Please do not duplicate the description. You can just redirect the reader to the header for it.

Reply:
Done